src/views/projectExplorer/index.ts

```diff
@@ -177,8 +177,30 @@ export default class ProjectExplorer implements TreeDataProvider<ProjectExplorer
           }
         }
       }),
-      commands.registerCommand(`vscode-ibmi-projectexplorer.assignToVariable`, async (element: TreeItem) => {
+      commands.registerCommand(`vscode-ibmi-projectexplorer.assignToVariable`, async (element: any) => {
+        if (element) {
+          const value = element.name ? element.name : element.path;
+          if (value) {
+            const variableItems: QuickPickItem[] = [];
+            const activeProject = ProjectManager.getActiveProject();
+            if (activeProject) {
+              const variables = await activeProject?.getVariables();
+              const values = await activeProject.getEnv();
+
+              for (const variable of variables) {
+                variableItems.push({ label: `&${variable}`, description: values[variable] });
+              }
+
+              const variable = await window.showQuickPick(variableItems, {
+                placeHolder: l10n.t('Select a variable')
+              });
 
+              if (variable) {
+                await activeProject.setEnv(variable.label.substring(1), value);
+              }
```

Comment:
what if I want to assign to a variable that does not exist yet?

Reply:
The `Assign to Variable` action is used to assign a value to an existing variable. `Configure as Variable` is used to configure a hardcoded value to a variable that the user can enter.

In what scenario would a user want to assign to a variable that does not exist in iproj.json?

Reply:
I see your point.  WIthout being referenced in the iproj.json, we dont have a variable defined.
The scenario I am trying to enable is to build the iproj.json without typing.
1) Go to QSYS browser and select s library as set as curlib or objlib or pre/post usrlibl
 -> has the effect of updating the iproj.json with hardcoded value
2) Go to Project Explorere and `Configure as Variable` (this part is already done)

So for step 1 I will create a separate issue.  We can consider this resolved.